### DISCUSSION TITLE
ESP32 WROOM - pins

### DISCUSSION THREAD

**Discussion by martelnov** [tobozo/ESP32-USB-Soft-Host#41] [Q&A]
Maybe stupid question... GPIOs 16 and 17 must be always enabled, or can be disabled and I can use other. On OLIMEX modules gpio17 is used for ethernet.

**Comment by tobozo** [comment]
DP_P0 and  DM_P0 are the only non-optional pins, their value can be 22,23 or 18,19 or any pair you test as valid.

Note: the "distance" between D+/D- pin numbers must be less than 8

**Comment by montalbinistefano** [comment]
hello, I'm trying to use an "olimex esp32 poe ISO" too, with pin 15-16 or 14-15 (and many other combination), but  I did not succeed, same sketch (USB_test) work fine with a node mcu esp32 (olimex is: esp32 wroom 32e, nodemcu is: esp32 wroom 32) any suggestion?
Thank you

**Comment by tobozo** [comment]
@montalbinistefano according to their [GPIO Map](https://www.olimex.com/Products/IoT/ESP32/ESP32-POE-ISO/resources/ESP32-POE-ISO-GPIO.png) pins 13-14-15 are share with the SD, many other pins are shared, either with a connector or a device

the only "free to use" pins on this card are 0, 1, 3, 32, 33, 35

**Reply by montalbinistefano** [reply]
thank you very much! i will try these.
